samples/sample_encode/src/pipeline_encode.cpp

```diff
@@ -463,8 +463,13 @@ mfxStatus CEncodingPipeline::InitMfxEncParams(sInputParams *pInParams)
 
     m_mfxEncParams.mfx.NumSlice = pInParams->nNumSlice;
     ConvertFrameRate(pInParams->dFrameRate, &m_mfxEncParams.mfx.FrameInfo.FrameRateExtN, &m_mfxEncParams.mfx.FrameInfo.FrameRateExtD);
-    m_mfxEncParams.mfx.EncodedOrder            = 0; // binary flag, 0 signals encoder to take frames in display order
 
+    m_mfxEncParams.mfx.EncodedOrder = pInParams->QPFileMode ? 1 : 0; // binary flag, 0 signals encoder to take frames in display order
+    if (m_mfxEncParams.mfx.EncodedOrder)
```

Comment:
here and everywhere further you use ```m_mfxEncParams.mfx.EncodedOrder``` to indicate that you should read QPs from file. But that's actually bad design, it is possible to use encoded order without manual QP settings... I suggest to have another class variable ```CEncodingPipeline:: bool QPFileMode;``` and use it instead of ```m_mfxEncParams.mfx.EncodedOrder```

Reply:
changed to m_bQPFileMode